File: src/roar_net_api/operations/lower_bound.py
# SPDX-FileCopyrightText: © 2025 Authors of the roar-net-api-py project <https://github.com/roar-net/roar-net-api-py/blob/main/AUTHORS>
#
# SPDX-License-Identifier: Apache-2.0

from typing import Optional, Protocol, TypeVar


_TValue_co = TypeVar("_TValue_co", covariant=True)


class SupportsLowerBound(Protocol[_TValue_co]):
    def lower_bound(self) -> Optional[_TValue_co]: ...

File: julia/src/operations/lower_bound.jl
# SPDX-FileCopyrightText: © 2025 Authors of the roar-net-api-py project <https://github.com/roar-net/roar-net-api-py/blob/main/AUTHORS>
#
# SPDX-License-Identifier: Apache-2.0
"""
    lower_bound(solution::Solution) -> value::Union{Nothing, <:Real}

Return a lower bound on the objective value of `solution`.
Used by beam search and other constructive algorithms.
"""
function lower_bound(::Solution)::Union{Nothing, <:Real} end
